File: SIN BDD/SOAP_JAVA_CONUNI_GR06/08. SERVIDOR/servidor_soap_java_conuni_gr06/test/ec/edu/monster/prueba/pruebaMasa.java
package ec.edu.monster.prueba;

import ec.edu.monster.servicio.ServicioMasa;
import org.junit.Test;
import static org.junit.Assert.*;

public class pruebaMasa {

    private final ServicioMasa servicio = new ServicioMasa();
    private static final double MARGEN = 0.0001;

    @Test
    public void pruebaKilogramosALibras() {
        assertEquals(2.20462, servicio.kilogramosALibras(1.0), MARGEN);
    }

    @Test
    public void pruebaGramosAOnzas() {
        assertEquals(3.5274, servicio.gramosAOnzas(100.0), MARGEN);
    }

    @Test
    public void pruebaToneladasAKilogramos() {
        assertEquals(2000.0, servicio.toneladasAKilogramos(2.0), MARGEN);
    }

    @Test
    public void pruebaLibrasAOnzas() {
        assertEquals(32.0, servicio.librasAOnzas(2.0), MARGEN);
    }

    @Test
    public void pruebaMiligramosAGramos() {
        assertEquals(1.0, servicio.miligramosAGramos(1000.0), MARGEN);
    }

    @Test
    public void pruebaValorCero() {
        assertEquals(0.0, servicio.kilogramosALibras(0.0), MARGEN);
    }
}

File: SIN BDD/SOAP_DOTNET_CONUNI_GR06/08. SERVIDOR/servidor_soap_dotnet_conuni_gr06/Pruebas/PruebaMasa.cs
using Ec.Edu.Monster.Controlador;

namespace Ec.Edu.Monster.Pruebas;

static class PruebaMasa
{
    static readonly CONUNI servicio = new();
    static int ok = 0, fail = 0;

    public static void Ejecutar()
    {
        Console.WriteLine("=== PRUEBAS DE CONVERSION DE MASA ===");
        Console.WriteLine();

        Probar("kilogramosALibras", 1, 2.20462);
        Probar("gramosAOnzas", 100, 3.527396);
        Probar("toneladasAKilogramos", 2, 2000);
        Probar("librasAOnzas", 1, 16);
        Probar("miligramosAGramos", 500, 0.5);

        Console.WriteLine();
        Console.WriteLine($"--- RESULTADOS MASA: OK={ok} FAIL={fail} ---");
        Console.WriteLine();
    }

    static void Probar(string operacion, double valor, double esperado)
    {
        try
        {
            double obtenido = operacion switch
            {
                "kilogramosALibras" => servicio.KilogramosALibras(valor),
                "gramosAOnzas" => servicio.GramosAOnzas(valor),
                "toneladasAKilogramos" => servicio.ToneladasAKilogramos(valor),
                "librasAOnzas" => servicio.LibrasAOnzas(valor),
                "miligramosAGramos" => servicio.MiligramosAGramos(valor),
                _ => throw new ArgumentOutOfRangeException(nameof(operacion))
            };

            bool casoOk = Math.Abs(obtenido - esperado) <= ConstantesPruebas.Tolerancia;

            if (casoOk)
            {
                ok++;
                Console.WriteLine($"[OK] {operacion}({valor}) = {obtenido:F4}");
            }
            else
            {
                fail++;
                Console.WriteLine($"[FAIL] {operacion}({valor})");
                Console.WriteLine($"  Esperado: {esperado} +/- {ConstantesPruebas.Tolerancia}");
                Console.WriteLine($"  Obtenido: {obtenido}");
            }
        }
        catch (Exception e)
        {
            fail++;
            Console.WriteLine($"[FAIL] {operacion}({valor}) -> Error: {e.Message}");
        }
    }
}
